code/io/joy-sdl.cpp

```diff
@@ -236,7 +239,7 @@ bool device_event_handler(const SDL_Event &evt)
 			}
 
 			// Add the new device to our list
-			auto added = JoystickPtr(new Joystick(device));
+			auto added = JoystickPtr(new Joystick(joyDeviceEvent.which));
```

Comment:
Might be good for debugging if the new joystick info was printed to log, instead of just at startup.